src/ImageSharp/Memory/DiscontiguousBuffers/MemoryGroup{T}.Owned.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SixLabors.ImageSharp.Memory
+{
+    // Analogous to the "owned" variant of MemorySource
+    internal abstract partial class MemoryGroup<T>
+    {
+        private class Owned : MemoryGroup<T>
+        {
+            private IMemoryOwner<T>[] memoryOwners;
+
+            public Owned(IMemoryOwner<T>[] memoryOwners, int bufferLength, long totalLength, bool swappable)
+                : base(bufferLength, totalLength)
+            {
+                this.memoryOwners = memoryOwners;
+                this.Swappable = swappable;
+                this.View = new MemoryGroupView<T>(this);
+            }
+
+            public bool Swappable { get; }
+
+            public override int Count
+            {
+                get
+                {
+                    this.EnsureNotDisposed();
+                    return this.memoryOwners.Length;
+                }
+            }
+
+            public override Memory<T> this[int index]
+            {
+                get
+                {
+                    this.EnsureNotDisposed();
+                    return this.memoryOwners[index].Memory;
+                }
+            }
+
+            public override IEnumerator<Memory<T>> GetEnumerator()
+            {
+                this.EnsureNotDisposed();
+                return this.memoryOwners.Select(mo => mo.Memory).GetEnumerator();
+            }
+
+            public override void Dispose()
```

Comment:
It's generally wise to add an internal check when disposing to ensure calling `Dispose()` multiple times cannot cause issues.

Reply:
This is done by checking `this.memoryOwners == null`. Could be more explicit by adding a property though.